### PR TITLE
Support git hooks

### PR DESCRIPTION
**Pull request by kylef**
This patch only adds pre and post commit hooks.

### CODE REVIEW

**Comment by technoweenie** [comment]
Thanks for this.  We'll have to do some testing on GitHub though.  I have a feeling this will cause hooks to fire twice on any repos or wikis edited through the web interface.  Since hooks weren't being fired, we'd manually queue up the internal Resque job that the actual hooks were supposed to queue.

**Comment by milki** [comment]
Has this been tested?

**Comment by dethstryke** [comment]
Is there any plan on taking this pull request?  Trying to get Gollum to sync with another repo.  This would help a lot.

**Comment by arr2036** [comment]
+1 if the code isn't too horrible ;)

**Comment by kylef** [comment]
@technoweenie, not to sure how GitHub fires its commits. But perhaps GitHub could have its own implementation of `Repo.commit` which will queue the jobs. Not sure where GitHub has currently put the code for the hooks, but it would make sense to fire all the hooks from the same place (both in grit and at github).

**Comment by clowder** [comment]
:+1:

Got here via github/gollum#69

**Comment by coffeencoke** [comment]
+1

File: lib/grit/index.rb [hunk 1]
```diff
@@ -97,6 +97,9 @@ def read_tree(tree)
     #
     # Returns a String of the SHA1 of the new commit.
     def commit(message, parents = nil, actor = nil, last_tree = nil, head = 'master')
+      hook = self.repo.hook('pre-commit')
+      return false unless hook && hook.success?
```
Comment (kainosnoema) [inline lib/grit/index.rb:101]: This aborts the commit if there's no `pre-commit` hook. Something like this is probably what you were intending:

``` rb
return false if hook && !hook.success?
```

**Comment by johnrfrank** [comment]
+1

**Comment by kylef** [comment]
Closing, this project has died since #183.

**Pull request closed without merging**